laws/src/main/scala/cats/laws/discipline/Arbitrary.scala

```diff
@@ -2,48 +2,77 @@ package cats
 package laws
 package discipline
 
+import scala.util.{Try, Success, Failure}
+
 import cats.data._
-import org.scalacheck.{Arbitrary, Gen}
+import org.scalacheck.{Arbitrary, Cogen, Gen}
 import org.scalacheck.Arbitrary.{arbitrary => getArbitrary}
 
 /**
  * Arbitrary instances for cats.data
  */
 object arbitrary extends ArbitraryInstances0 {
 
-  // A special function1Arbitrary for testing operations like dropWhile specifically
-  // in the context of Int => Boolean. Once scalacheck supports better Function1 arbitrary
-  // instances this can be removed.
-  implicit def catsLawsArbitraryForIntToBool: Arbitrary[(Int) => Boolean] =
-    Arbitrary(
-      getArbitrary[Int].map(x =>
-       new Function1[Int, Boolean] {
-         def apply(i: Int): Boolean = i < x
+  implicit def catsLawsCogenForStream[A: Cogen]: Cogen[Stream[A]] =
+    Cogen[List[A]].contramap(_.toList)
+
+  implicit val catsLawsCogenForThrowable: Cogen[Throwable] =
+    Cogen[String].contramap(_.toString)
 
-         override def toString = s"<function testing whether input is less than $x>"
-       }))
+  implicit def catsLawsCogenForTry[A](implicit A: Cogen[A]): Cogen[Try[A]] =
+    Cogen((seed, x) => x match {
+      case Success(a) => A.perturb(seed, a)
+      case Failure(e) => Cogen[Throwable].perturb(seed, e)
+    })
+
+  implicit def catsLawsCogenForFunction0[A](implicit A: Cogen[A]): Cogen[Function0[A]] =
+    A.contramap(_())
```

Comment:
A lot of these instances also look like they are defined in scalacheck. Same question as https://github.com/typelevel/cats/pull/1345/files#r77895807